pageserver/src/tenant/timeline.rs

```diff
@@ -2076,6 +2077,18 @@ impl Timeline {
             result.layers_removed += 1;
         }
 
+        info!(
+            "GC completed removing {} layers, cuttof {}",
+            result.layers_removed, new_gc_cutoff
+        );
+        if result.layers_removed != 0 {
+            fail_point!("gc-before-save-metadata", |_| {
+                info!("Abnormaly terinate pageserver at gc-before-save-metadata fail point");
+                std::process::abort();
+            });
+            return Ok(result);
+        }
```

Comment:
Hang on, why is the `return Ok(result)` here? So if we decided to remove any layers, we do a quick exit, and skip actually scheduling the deletions in remote storage?

Reply:
Opened https://github.com/neondatabase/neon/pull/2655 for that

Reply:
Sorry, I dont't remember the reasons of adding this early return.
Looks like it is result of some my experiments when I tried to reproduce the problem.
It is definitely a bug thank you for fixing it.